federationapi/federationapi_test.go

```diff
@@ -362,3 +368,118 @@ func TestRoomsV3URLEscapeDoNot404(t *testing.T) {
 		}
 	}
 }
+
+func TestNotaryServer(t *testing.T) {
+
+	// Start a server we are going to request keys from
+	testCases := []struct {
+		name          string
+		httpBody      string
+		pubKeyRequest *gomatrixserverlib.PublicKeyNotaryLookupRequest
+		validateFunc  func(t *testing.T, response util.JSONResponse)
+	}{
+		{
+			name: "empty httpBody",
+			validateFunc: func(t *testing.T, resp util.JSONResponse) {
+				want := util.JSONResponse{
+					Code: http.StatusBadRequest,
+					JSON: spec.BadJSON("The request body could not be decoded into valid JSON. unexpected end of JSON input"),
```

Comment:
It would be nicer to assert the error code and not the entire error message, as that is pretty brittle.